src/main/java/org/mastodon/mamut/tomancak/merging/Dataset.java

```diff
@@ -47,26 +44,30 @@ public class Dataset
 {
 	private final MamutProject project;
 
-	private final int numTimepoints;
-
 	private final Model model;
 
 	private final int maxNonEmptyTimepoint;
 
 	public Dataset( final String path ) throws IOException
 	{
 		project = MamutProjectIO.load( path );
-		numTimepoints = getNumTimepoints( project );
 		model = new Model();
 		try (final MamutProject.ProjectReader reader = project.openForReading())
 		{
 			model.loadRaw( reader );
 		}
-		maxNonEmptyTimepoint = getMaxNonEmptyTimepoint( model, numTimepoints );
-
+		maxNonEmptyTimepoint = maxTimepoint( model );
 		verify();
 	}
 
+	private int maxTimepoint( Model model )
```

Comment:
This method already existed in the deep lineage plugin. In order to reduce code duplication, I created a PR on mastodon-core that contains these methods: https://github.com/mastodon-sc/mastodon/pull/274
I suggest to remove this method, after the PR on the core has been merged.

Reply:
Yes it's good to have this method in mastodon core